caas/kubernetes/provider/k8s.go

```diff
@@ -458,11 +503,19 @@ func (k *kubernetesClient) EnsureService(
 		}
 	}()
 
-	unitSpec, err := makeUnitSpec(params.PodSpec)
+	unitSpec, err :=
+		makeUnitSpec(appName, params.PodSpec)
 	if err != nil {
 		return errors.Annotatef(err, "parsing unit spec for %s", appName)
 	}
 
+	for _, c := range params.PodSpec.Containers {
+		if err := k.EnsureSecret(appName, &c); err != nil {
```

Comment:
check here that image details has a password and skip if not
then have EnsureSecret return an error if no password